src/AasxServerDB/VisitorAASX.cs

```diff
@@ -0,0 +1,478 @@
+﻿using AasCore.Aas3_0;
+using AdminShellNS;
+using System.Globalization;
+using static AasCore.Aas3_0.Visitation;
+using Extensions;
+using System.IO.Compression;
+
+namespace AasxServerDB
+{
+    public class VisitorAASX : VisitorThrough
+    {
+        AASXSet _aasxDB;
+        SMSet _smDB;
+        SMESet _parSME = null;
+
+        public VisitorAASX(AASXSet? aasxDB = null)
+        {
+            _aasxDB = aasxDB;
+        }
+
+        static public void LoadAASXInDB(string filePath, bool createFilesOnly, bool withDbFiles)
+        {
+            using (var asp = new AdminShellPackageEnv(filePath, false, true))
+            {
+                if (!createFilesOnly)
+                {
+                    var aasxDB = new AASXSet
+                    {
+                        AASX = filePath
+                    };
+                    LoadAASInDB(asp, aasxDB);
+
+                    using AasContext db = new AasContext();
+                    db.Add(aasxDB);
+                    db.SaveChanges();
+                }
+
+                if (withDbFiles)
+                {
+                    string name = Path.GetFileName(filePath);
+                    try
+                    {
+                        string fcopyt = name + "__thumbnail";

```

Comment:
Did you rather mean something like this?

```suggestion
                        string temporaryFileName = name + "__thumbnail";

```